packages/es-components/src/components/controls/buttons/IconButton.js

```diff
@@ -2,133 +2,129 @@ import React from 'react';
+  max-width: ${props => props.maxWidth};
+`;
 
-  &:hover:enabled:not(:focus),
-  &:focus:enabled {
-    box-shadow: 0 0 0 0.2rem ${props => props.theme.brandColors.primary3};
-    outline: 0;
-  }
+const StyledIconButton = styled(OutlineButton)`
+  background-color: ${props => props.isHighlighted && props.activeColor};
+  border-radius: 50%;
+  border-width: 3px;
+  color: ${props => props.isHighlighted && props.theme.colors.white};
+  display: inline-block;
+  font-weight: normal;
+  min-width: 0;
+  padding: 13px 14px;
+  width: auto;
+`;
+
+const IncompleteButton = styled(StyledIconButton)`
+  background-color: ${props =>
+    props.isHighlighted ? props.theme.colors.gray5 : props.theme.colors.white};
+  border-left: 3px dashed ${props => props.theme.colors.gray5};
+  color: ${props =>
+    props.isHighlighted ? props.theme.colors.white : props.theme.colors.gray5};
+  transform: rotate(45deg);
 
-  &:active:enabled ${IconWithBorder} {
-    background-color: ${props => props.theme.brandColors.primary3};
+  &:hover,
+  &:active {
+    background-color: ${props => props.theme.colors.gray6};
     color: ${props => props.theme.colors.white};
   }
-  &:active:enabled ${HighlightedIconWithBorder} {
-    background-color: ${props => props.theme.colors.white};
-    color: ${props => props.theme.brandColors.primary3};
-  }
 
-  &:active:enabled ${ChildrenSpan} {
-    font-weight: bold;
-  }
-  &:active:enabled ${HighlightedChildrenSpan} {
-    font-weight: normal;
+  & i {
+    transform: rotate(-45deg) translate(-1px, 1px);
   }
 `;
 
+const IconText = styled.span`
+  font-size: ${props => props.fontSize}px;
+  font-weight: ${props => (props.isHighlighted ? 'bold' : 'normal')};
+  margin-top: 6px;
+  max-width: ${props => props.maxWidth};
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`;
+
 function IconButton({
   iconName,
   iconSize,
   childrenFontSize,
   isHighlighted,
+  isIncomplete,
   disabled,
   maxWidth,
   onClick,
-  borderRadiiSize,
+  styleType,
   children,
   ...otherProps
 }) {
-  const CurrentIcon = isHighlighted
-    ? HighlightedIconWithBorder
-    : IconWithBorder;
-  const CurrentSpan = isHighlighted ? HighlightedChildrenSpan : ChildrenSpan;
+  const CurrentButton = isIncomplete ? IncompleteButton : StyledIconButton;
+  const theme = useTheme();
+  const variant = theme.buttonStyles.outlineButton.variant[styleType];
 
   return (
-    <ContainerButton
-      onClick={onClick}
-      isHighlighted={isHighlighted}
-      disabled={disabled}
-      borderRadii={borderRadiiSize}
-      {...otherProps}
-    >
-      <CurrentIcon
-        name={iconName}
-        size={iconSize}
+    <Container maxWidth={maxWidth} {...otherProps}>
+      <CurrentButton
+        onClick={onClick}
+        styleType={styleType}
         isHighlighted={isHighlighted}
-      />
-      <CurrentSpan
+        disabled={disabled}
+        activeColor={variant.bgColor}
+      >
+        <Icon name={iconName} size={iconSize} />
```

Comment:
A thought struck me this morning: do we need something in addition to this Icon so screen readers can report the purpose of this button? Maybe some text we hide off screen?

Reply:
I think the `IconText` should be enough. Screen Readers will say something like "Button, <Name>".

Reply:
Would it be worth adding an `aria-describedby` to the button that links to a generated id on the `IconText`?